### DISCUSSION TITLE
Use AGPL rather than GPL?

### DISCUSSION THREAD

**Discussion by TheTechRobo** [OpenShopChannel/WSC-Web#18] [General]
If I'm correct, the plain GPL has a problem for server applications.

Cf. https://www.gnu.org/licenses/why-affero-gpl.en.html:

> Suppose you develop and release a free program under the ordinary GNU GPL. If developer D modifies the program and releases it, the GPL requires him to distribute his version under the GPL too. Thus, if you get a copy of his version, you are free to incorporate some or all of his changes into your own version.
But suppose the program is mainly useful on servers. When D modifies the program, he might very likely run it on his own server and never release copies. Then you would never get a copy of the source code of his version, so you would never have the chance to include his changes in your version. [...]
Using the GNU Affero GPL avoids that outcome. [...] Assuming he has followed the license requirement to let the server's users download the source code of his version, you can do so, and then you can incorporate his changes into your version.

I believe this also affects this program, which can allow people to circumvent the GPL's requirements. (I could be wrong, though.)

**Comment by dhtdht020** [comment]
I will have to investigate this further, thanks for pointing it out.
